credentials/credentials.go

```diff
@@ -126,6 +126,10 @@ func (t TLSInfo) AuthType() string {
 	return "tls"
 }
 
+func (t TLSInfo) ConnState() tls.ConnectionState {
```

Comment:
> Making it exported also makes a copy when users work on it (notice
tls.ConnectionState is a object not a pointer).

Doesn't matter that it's not a pointer. If it's exported I can overwrite it entirely: `tlsInfo.State = myOtherTlsState`.

> Plus, making a copy does
NOT prohibit mutation because tls.ConnectionState contains a number of
pointer fields.

Yes, you could mess with some of the slices in [tls.ConnectionState](http://golang.org/pkg/crypto/tls/#ConnectionState) but it's still better than exporting this.

> Actually adding a getter might make the 2nd copy (depends on how the
compiler optimizes it) which is not needed.

This doesn't matter - those copies are on the stack and the Go compiler does RVO.